orca-pipelinetemplate/src/main/java/com/netflix/spinnaker/orca/pipelinetemplate/v1schema/handler/v2/V2TemplateLoaderHandler.java

```diff
@@ -0,0 +1,126 @@
+import com.netflix.spinnaker.orca.pipelinetemplate.v1schema.model.TemplateConfiguration;
+import com.netflix.spinnaker.orca.pipelinetemplate.v1schema.render.RenderContext;
+import com.netflix.spinnaker.orca.pipelinetemplate.v1schema.render.v2.V2DefaultRenderContext;
+import com.netflix.spinnaker.orca.pipelinetemplate.v1schema.render.v2.V2RenderUtil;
+import com.netflix.spinnaker.orca.pipelinetemplate.v2schema.model.V2PipelineTemplate;
+
+import javax.validation.constraints.NotNull;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+public class V2TemplateLoaderHandler implements Handler {
+
+  private static final TypeReference<Map<String, Object>> MAP_TYPE_REFERENCE = new TypeReference<Map<String, Object>>() {};
+  private V2TemplateLoader templateLoader;
+  private ContextParameterProcessor contextParameterProcessor;
+  private ObjectMapper objectMapper;
+
+  public V2TemplateLoaderHandler(V2TemplateLoader templateLoader, ContextParameterProcessor contextParameterProcessor, ObjectMapper objectMapper) {
+    this.templateLoader = templateLoader;
+    this.contextParameterProcessor = contextParameterProcessor;
+    this.objectMapper = objectMapper;
+  }
+
+  @Override
+  public void handle(@NotNull HandlerChain chain, @NotNull PipelineTemplateContext context) {
+    TemplateConfiguration config = objectMapper.convertValue(context.getRequest().getConfig(), TemplateConfiguration.class);
+
+    // Allow template inlining to perform plans without publishing the template
+    if (context.getRequest().getPlan() && context.getRequest().getTemplate() != null) {
+      V2PipelineTemplate template = objectMapper.convertValue(context.getRequest().getTemplate(), V2PipelineTemplate.class);
+      context.setSchemaContext(new V2PipelineTemplateContext(config, template));
+      return;
+    }
+
+    Map<String, Object> trigger = context.getRequest().getTrigger();
+    // Allow the config's source to be dynamically resolved from trigger payload.
+    renderPipelineTemplateSource(config, trigger);
+
+    // If a template source isn't provided by the configuration, we're assuming that the configuration is fully-formed.
+    V2PipelineTemplate template = new V2PipelineTemplate();
+    if (config.getPipeline().getTemplate() == null) {
+      List<V2PipelineTemplate.Variable> variables = config.getPipeline().getVariables().entrySet().stream()
+        .map(v -> {
+          return V2PipelineTemplate.Variable.builder().name(v.getKey()).defaultValue(v.getValue()).build();
+        })
+        .collect(Collectors.toList());
+      template.setVariables(variables);
+    } else {
+      template = templateLoader.load(config.getPipeline().getTemplate());
+    }
+
+    // Ensure that any expressions contained with template variables are rendered
+    // TODO(jacobkiefer): This nests application, trigger, and pipelineTemplate under a 'variables'
+    // namespace in the SpEL processing. Should we un-nest this?
```

Comment:
Are these nested only during the first evaluation, or during the execution?

Reply:
So far this is contained in the first evaluation. I'm not planning to carry this over to the execution resolution. The nesting feels clunky to me since you'd refer to `trigger` attributes by doing `${ variables.trigger.field }` which doesn't feel right.